From: Xinmu <xinmu@loongfans.cn>
Subject: Fix Navi12 Crash

Navi12 # cat /sys/class/kfd/kfd/topology/nodes/1/properties
 max_waves_per_simd 20
Navi48/44/20/Raphael # cat /sys/class/kfd/kfd/topology/nodes/1/properties
 max_waves_per_simd 16

---
diff --git a/libhsakmt/src/topology.c b/libhsakmt/src/topology.c
index cc33c738f5f..aac8eace078 100644
--- a/libhsakmt/src/topology.c
+++ b/libhsakmt/src/topology.c
@@ -1267,6 +1267,11 @@ static HSAKMT_STATUS topology_sysfs_get_node_props(HsaKFDContext *ctx,
 			gfxv = (uint32_t)prop_val;
 	}
 
+	// Navi12 send max_waves_per_simd: 20, should be 16.
+	if (gfxv == 100101)
+		if (props->MaxWavesPerSIMD == 20)
+			props->MaxWavesPerSIMD = 16;
+
 	if (!ctx->hsakmt_is_svm_api_supported)
 		props->Capability.ui32.SVMAPISupported = 0;
 
